### PR TITLE
Adding info about the fiveg_n4 interface

### PR DESCRIPTION
**Pull request by Gmerold**
**Overview:**
In 5G Core, the `N4` interface is used to establish a bridge between the User Plane and the Control Plane. 
A new charm relation interface facilitates connectivity over the N4 interface, between the interface provider and requirer.

**Reference:**
https://docs.google.com/document/d/1s3U82GTUZFJ9mEhiHig0oAl44F0fq3eQT2PNv1uV75g/edit

### CODE REVIEW

File: interfaces/fiveg_n4/v0/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+## Usage
+
+Within 5G, the User Plane Function (UPF) supports features and capabilities to facilitate 
+user plane operation. Examples include: packet routing and forwarding, interconnection 
+to the Data Network, policy enforcement and data buffering.
+
+The `fiveg_n4` relation interface describes the expected behavior of any charm claiming to be able 
+to provide or consume the UPF information in order to establish communication over the N4 interface. 
+In a typical 5G network, the N4 interface is used between UPF and AMF.
```
[the remark this answers is not in the document]
Comment (ghislainbourgeois) [reply interfaces/fiveg_n4/v0/README.md:11]: In our specific case, this relation will not be consumed directly by the SMF, but probably by the new NMS to allow it to be configured in a network slice. I am not sure if we want to add a note on that.

**Review by ghislainbourgeois — CHANGES_REQUESTED** [review]

**Review by gruyaume — APPROVED** [review]
LGTM besides @ghislainbourgeois 's comments

File: interfaces/fiveg_n4/v0/schema.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    ProviderSchema:
+        unit: <empty>
+        app: {
+            "upf_hostname": "upf.uplane-cloud.canonical.com",
+            "upf_port": 8805
+        }
+    RequirerSchema:
+        unit: <empty>
+        app:  <empty>
+"""
```
Comment (DanielArndt) [inline interfaces/fiveg_n4/v0/schema.py:15]:
```suggestion
"""This file defines the schemas for the provider and requirer sides of the `fiveg_n4` interface.

It exposes two `interfaces.schema_base.DataBagSchema` subclasses called:
- `ProviderSchema`
- `RequirerSchema`

Examples:

    ProviderSchema:
        unit: <empty>
        app: {
            "upf_hostname": "upf.uplane-cloud.canonical.com",
            "upf_port": 8805
        }

    RequirerSchema:
        unit: <empty>
        app:  <empty>
"""
```

File: interfaces/fiveg_n4/v0/README.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+and a Requirer.
+
+## Behavior
+
+Both the Requirer and the Provider need to adhere to criteria to be considered compatible 
+with the interface.
+
+### Provider
+
+- Is expected to provide the hostname and the port of the UPF's N4 interface.
```
[the remark this answers is not in the document]
Comment (Gmerold) [reply interfaces/fiveg_n4/v0/README.md:30]: The protocol for the N4 interface is specified by 3GPP, so I figured we don't have mention it here.

File: interfaces/fiveg_n4/v0/schema.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+            "upf_hostname": "upf.uplane-cloud.canonical.com",
+            "upf_port": 8805
+        }
+
+    RequirerSchema:
+        unit: <empty>
+        app:  <empty>
+"""
+
+from pydantic import BaseModel, IPvAnyAddress, Field
```
Comment (patriciareinoso) [inline interfaces/fiveg_n4/v0/schema.py:21]:
```suggestion
from pydantic import BaseModel, Field
```

File: interfaces/fiveg_n4/v0/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ghislainbourgeois) [reply interfaces/fiveg_n4/v0/README.md]: I agree that mentioning the NMS is a bit too specific to this one implementation. I think the wording here is fine.

**Review by ghislainbourgeois — APPROVED** [review]

**Review by simskij — APPROVED** [review]

**Pull request merged**